services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -1972,6 +1986,15 @@ public void rollForwardToFutureVersion(String clusterName, String storeName, Str
       message.payloadUnion = rollForwardCurrentVersion;
 
       sendAdminMessageAndWaitForConsumed(clusterName, storeName, message);
+      Map<String, String> futureVersions = getFutureVersionsForMultiColos(clusterName, storeName);
+      int futureVersion = 0;
+      for (Map.Entry<String, String> entry: futureVersions.entrySet()) {
+        futureVersion = Integer.parseInt(entry.getValue());
+        if (futureVersion > 0) {
+          break;
+        }
+      }
+      truncateKafkaTopic(Version.composeKafkaTopic(storeName, futureVersion));
```

Comment:
If child controllers manage to promote the version quickly, getFutureVersion API might return nothing?

If the above race condition could happen, can we get the list of future versions first, send admin message next, and eventually truncation the parent topics?
Besides, why there could be more than 1 future version number? In theory, there is only one topic?

Reply:
Good point, I will swap the order of the operations. Ideally there should not be more than 1 future version, but for  this API is for displaying future version per colo, which can have no future version for certain colo.